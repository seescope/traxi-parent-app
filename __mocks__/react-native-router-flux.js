import React, { PropTypes } from 'react';
import { View } from 'react-native';

export const Actions = {
  createKid: jest.fn(),
  walkthrough: jest.fn(),
  congratulations: jest.fn(),
  thankyou: jest.fn(),
  areYouReady: jest.fn(),
  intro: jest.fn(),
};

export const Scene = (props) => <View {...props} />;

export const Router = ({ children }) => (
  <View>{children}</View>
);

Router.propTypes = {
  children: PropTypes.array.isRequired,
};
