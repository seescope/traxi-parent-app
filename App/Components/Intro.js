import React from 'react';
import Spacing from '../Components/Spacing';
import { Actions } from 'react-native-router-flux';
import {
  Animated,
  TouchableOpacity,
  Platform,
  View,
  Image,
  Dimensions,
  Text,
} from 'react-native';
import * as Animatable from 'react-native-animatable';
import Analytics from 'react-native-analytics';

import ProgressTrack from '../Components/ProgressTrack';
import { WHITE, TRANSPARENT } from '../Constants/Colours';
import Button from '../Components/Button';

const containerStyle = {
  flex: 1,
};

const ANIMATION_DURATION = 500;

const { width, height } = Dimensions.get('window');
const imageStyle = {
  position: 'absolute',
  width,
  height,
};

const headerStyle = {
  backgroundColor: TRANSPARENT,
  color: WHITE,
  fontSize: Platform.select({
    ios: 28,
    android: 24,
  }),
  textAlign: 'center',
  fontFamily: 'Raleway-ExtraBold',
  marginHorizontal: 16,
  marginBottom: 16,
};

const subHeaderStyle = {
  backgroundColor: TRANSPARENT,
  color: WHITE,
  fontSize: Platform.select({
    ios: 16,
    android: 14,
  }),
  textAlign: 'center',
  fontFamily: 'Raleway-Regular',
  marginHorizontal: 16,
  marginBottom: 16,
};

const paddingStyle = {
  paddingHorizontal: 32,
  flexDirection: 'row',
  alignItems: 'center',
  justifyContent: 'space-between',
  flex: 5,
};

const textContainer = {
  alignItems: 'center',
  flex: 3,
};

const STEP_IMAGES = [
  require('../Images/intro-step-1.png'),
  require('../Images/intro-step-2.png'),
  require('../Images/intro-step-3.png'),
  require('../Images/intro-step-4.png'),
];

const HEADER_TEXT = [
  'First, get your kid\'s device',
  'Then enter a PIN',
  'Follow the instructions',
  'Then you\'re all done!',
];

const SUBHEADER_TEXT = [
  'Don\'t worry, you can do this later if you like.',
  'We\'ll tell you what it is soon',
  'It only takes a couple of seconds',
  'Doesn\'t that look easy?',
];

export const onPress = () => {
  Analytics.track('Finished Intro');
  Actions.areYouReady();
};

export default class extends React.Component {
  constructor() {
    super();

    this.state = {
      step: 0,
    };

    this.animations = [
      new Animated.Value(1),
      new Animated.Value(0),
      new Animated.Value(0),
      new Animated.Value(0),
    ];

    this.styles = [
      { opacity: this.animations[0] },
      { opacity: this.animations[1] },
      { opacity: this.animations[2] },
      { opacity: this.animations[3] },
    ];
  }

  nextStep() {
    const { step } = this.state;
    const newStep = step < 4 ? step + 1 : step;

    Animated.timing(
      this.animations[newStep],
      {
        toValue: 1,
        duration: ANIMATION_DURATION,
      }
    ).start();

    this.setState({
      step: newStep,
    });
  }

  previousStep() {
    const { step } = this.state;
    const newStep = step > 0 ? step - 1 : step;

    Animated.timing(
      this.animations[step],
      {
        toValue: 0,
        duration: ANIMATION_DURATION,
      }
    ).start();

    this.setState({
      step: newStep,
    });
  }

  render() {
    return (
      <View style={containerStyle}>
        <Animated.Image
          resizeMode="cover"
          style={[imageStyle, this.styles[0]]}
          source={STEP_IMAGES[0]}
        />
        <Animated.Image
          resizeMode="cover"
          style={[imageStyle, this.styles[1]]}
          source={STEP_IMAGES[1]}
        />
        <Animated.Image
          resizeMode="cover"
          style={[imageStyle, this.styles[2]]}
          source={STEP_IMAGES[2]}
        />
        <Animated.Image
          resizeMode="cover"
          style={[imageStyle, this.styles[3]]}
          source={STEP_IMAGES[3]}
        />

        <View style={paddingStyle}>
          {this.state.step > 0 ? <TouchableOpacity onPress={() => this.previousStep()}>
            <Image
              source={require('../Images/left-arrow.png')}
            />
          </TouchableOpacity> : <View />}
          {this.state.step < 3 ? <Animatable.View
            useNativeDriver
            animation="bounce"
            duration={2000}
            delay={3000}
          >
            <TouchableOpacity onPress={() => this.nextStep()}>
              <Image
                source={require('../Images/right-arrow.png')}
              />
            </TouchableOpacity>
          </Animatable.View> : <View />}
        </View>

        <View style={textContainer}>
          <Animatable.View
            useNativeDriver
            easing="ease-out"
            duration={1000}
            animation="bounceInLeft"
          >
            <ProgressTrack
              stage={this.state.step}
              width={width - 64}
            />
          </Animatable.View>

          <Spacing height={32} />

          <Text style={headerStyle}>{HEADER_TEXT[this.state.step]}</Text>

          <Text style={subHeaderStyle}>{SUBHEADER_TEXT[this.state.step]}</Text>

          <Spacing height={16} />

          {this.state.step === 3 && <Animatable.View
            useNativeDriver
            delay={1000}
            duration={1000}
            easing="ease-in-out"
            animation="bounceInUp"
          >
            <Button onPress={() => onPress()} primary={false}>Got it!</Button>
          </Animatable.View>}
        </View>
      </View>
    );
  }
}
