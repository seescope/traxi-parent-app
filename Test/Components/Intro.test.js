jest.mock('../../App/Components/ProgressTrack', () => () => null);

import React from 'react';
import { AsyncStorage } from 'react-native';
import renderer from 'react-test-renderer';
import { Actions } from 'react-native-router-flux';
import Intro, { onPress } from '../../App/Components/Intro';

it('renders', () => {
  const tree = renderer.create(<Intro />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('saves the parent\'s state and transitions to the next screen', () => {
  AsyncStorage.setItem = jest.fn(() => Promise.resolve());

  return onPress().then(() => {
    expect(Actions.areYouReady).toHaveBeenCalled();
    expect(AsyncStorage.setItem.mock.calls).toMatchSnapshot();
  });
});
