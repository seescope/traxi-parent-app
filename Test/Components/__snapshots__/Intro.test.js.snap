exports[`test renders 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }>
  <Image
    resizeMode="cover"
    source={1}
    style={
      Object {
        "height": 1334,
        "opacity": 1,
        "position": "absolute",
        "width": 750,
      }
    } />
  <Image
    resizeMode="cover"
    source={1}
    style={
      Object {
        "height": 1334,
        "opacity": 0,
        "position": "absolute",
        "width": 750,
      }
    } />
  <Image
    resizeMode="cover"
    source={1}
    style={
      Object {
        "height": 1334,
        "opacity": 0,
        "position": "absolute",
        "width": 750,
      }
    } />
  <Image
    resizeMode="cover"
    source={1}
    style={
      Object {
        "height": 1334,
        "opacity": 0,
        "position": "absolute",
        "width": 750,
      }
    } />
  <Image
    resizeMode="cover"
    source={1}
    style={
      Object {
        "height": 1334,
        "opacity": 0,
        "position": "absolute",
        "width": 750,
      }
    } />
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 5,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 32,
      }
    }>
    <View />
    <View
      animation="bounce"
      delay={3000}
      duration={2000}
      useNativeDriver={true}>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}>
        <Image
          source={1} />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 3,
      }
    }>
    <View
      animation="bounceInLeft"
      duration={1000}
      easing="ease-out"
      useNativeDriver={true} />
    <View
      style={
        Object {
          "height": 32,
        }
      } />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "backgroundColor": "transparent",
          "color": "#FFF",
          "fontFamily": "Raleway-ExtraBold",
          "fontSize": 28,
          "marginBottom": 16,
          "marginHorizontal": 16,
          "textAlign": "center",
        }
      }>
      [missing \"en.intro.header0\" translation]
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "backgroundColor": "transparent",
          "color": "#FFF",
          "fontFamily": "Raleway-Regular",
          "fontSize": 16,
          "marginBottom": 16,
          "marginHorizontal": 16,
          "textAlign": "center",
        }
      }>
      [missing \"en.intro.subheader0\" translation]
    </Text>
    <View
      style={
        Object {
          "height": 16,
        }
      } />
  </View>
</View>
`;

exports[`test sends tracking to Segment that intro was seen 1`] = `
Array [
  Array [
    "Finished Intro",
  ],
]
`;
